firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/Serial/Nonin/Sensor.cpp

```diff
@@ -14,65 +14,80 @@ namespace Pufferfish::Driver::Serial::Nonin {
 // Sensor
 
 InitializableState Sensor::setup() {
```

Comment:
We should return early if last state was not `InitializableState::setup`.

Reply:
If `prev_state_` is already `ok`, then we should do an early return of `ok`; if `prev_state_` is already `failed`, then we should do an early return of `failed`; if `prev_state_` is `setup`, then we should proceed to run the setup code. In other words, we can just do `if (prev_state_ != InitializableState::setup) {return prev_state_;}`.

The `Sensor` object starts out with `prev_state_` initialized as `setup`, so `prev_state_ != InitializableState::ok` when the firmware first starts. Your change below causes `Sensor::setup` to return `failed` the first time we call `setup`